jinahub/indexers/SimpleIndexer/__init__.py

```diff
@@ -44,60 +44,34 @@ def __init__(
         self._flush = True
         self._docs_embeddings = None
 
-    @property
-    def index_embeddings(self):
-        if self._flush:
-            self._docs_embeddings = np.stack(self._docs.get_attributes('embedding'))
-            self._flush = False
-        return self._docs_embeddings
-
     @requests(on='/index')
-    def index(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+    def index(self, docs: Optional['DocumentArray'] = None, parameters: Optional[Dict] = {}, **kwargs):
         """All Documents to the DocumentArray
         :param docs: the docs to add
         :param parameters: the parameters dictionary
         """
-        traversal_path = parameters.get('traversal_paths', self.default_traversal_paths)
-        flat_docs = docs.traverse_flat(traversal_path)
+        if not docs: return
+        traversal_paths = parameters.get('traversal_paths', self.default_traversal_paths)
+        flat_docs = docs.traverse_flat(traversal_paths)
         self._docs.extend(flat_docs)
         self._flush = True
 
     @requests(on='/search')
-    def search(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+    def search(self, docs: Optional['DocumentArray'] = None, parameters: Optional[Dict] = {}, **kwargs):
         """Perform a vector similarity search and retrieve the full Document match
 
         :param docs: the Documents to search with
         :param parameters: the parameters for the search"""
-        traversal_path = parameters.get('traversal_paths', self.default_traversal_paths)
+        if not docs or not self._docs: return
```

Comment:
if not self._docs at share time maybe it would be good to have some logs?

Reply:
Addressed, please see [here](https://github.com/jina-ai/executors/pull/47/commits/a154dffeefe6112776c32d6d0a206d163af1bf66) for the change.